docparse/jsonschema.go

```diff
@@ -208,10 +207,25 @@ func fieldToSchema(prog *Program, fName string, ref Reference, f *ast.Field) (*S
 start:
 	switch typ := sw.(type) {
 
-	// Don't support interface{} for now. We'd have to add a lot of complexity
-	// for it, and not sure if we're ever going to need it.
+	// Interface, only useful for its description.
 	case *ast.InterfaceType:
-		return nil, errors.New("fieldToSchema: interface{} is not supported")
+		field := f.Names[0].Obj.Decl.(*ast.Field)
```

Comment:
`len(f.Names)` can be zero:

```
type foo struct {
    io.Reader

    named string
}

func (f foo) Read(..) { .. }
```

Can just return an error in that case with `TODO` comment as far as I'm concerned.

Reply:
I did try embedding an interface and it didn't panic, can give it another go in morning or add it to test case since it shouldn't affect the output

Reply:
Added it to the test case, it doesn't seem possible to get here with a field that no names when embedding an interface. Added the error anyway in case there is some way to do it will make it easier to track down the issue.

Reply:
Cool, I just assumed it would error out. Thanks for adding test case!